src/main/java/frc/team2412/robot/util/MultiplexedColorSensor.java

```diff
@@ -0,0 +1,97 @@
+package frc.team2412.robot.util;
+
+import com.revrobotics.ColorSensorV3;
+import com.revrobotics.ColorSensorV3.RawColor;
+
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.util.Color;
+import frc.team2412.robot.Hardware.HardwareConstants;
+
+/**
+ * based on https://gist.github.com/SCOTSBots/4daf384311d32e9ff0aad2f179819139
+ * the reason why not extend or inherit the ColorSensorV3 is because
+ * in order to use any device attached to the Multiplexer have to setChannel() beforehand
+ * so if called any method that's not been override will cause trouble
+ */
+public class MultiplexedColorSensor {
+    private I2C i2c_multiplexer;
+    private final int i2c_multiplexer_address;
+    // The actual sensor. All of the methods call this sensor to get the data.
+    private final ColorSensorV3 sensor;
+    // What port on the multiplexer the color sensor is plugged into.
+    private final int sensor_port_number;
+  
```

Comment:
To ensure consistent naming conventions, consider using snake case (i.e., i2cMultiplexer and portNumber) instead of snake case (i2c_multiplexer and port_number).